src/SIL.XForge.Scripture/ClientApp/src/app/shared/share/share-dialog.component.ts

```diff
@@ -1,22 +1,256 @@
+    private readonly projectService: SFProjectService,
+    private readonly pwaService: PwaService,
+    private readonly userService: UserService
   ) {
-    dialogRef.addPanelClass('share-dialog-component-panel');
+    super();
+    this.projectId = this.data.projectId;
+    Promise.all([
+      this.projectService.getProfile(this.projectId),
+      this.projectService.isProjectAdmin(this.projectId, this.userService.currentUserId)
+    ]).then(value => {
+      this.projectDoc = value[0];
+      this.isProjectAdmin = value[1];
+      this.subscribe(this.projectDoc.remoteChanges$, () => {
+        if (this.shareLinkUsageOptions.length === 0) {
+          this.dialogRef.close();
+        } else if (!this.shareLinkUsageOptions.includes(this.shareLinkType)) {
+          this.resetLinkUsageOptions();
+        } else {
+          this.updateSharingKey();
+        }
+      });
+      this.shareRole = this.defaultShareRole;
+      if (this.isProjectAdmin) {
+        this.shareLinkType = ShareLinkType.Recipient;
+      }
+      this.subscribe(this.pwaService.onlineStatus$, () => this.updateSharingKey());
+    });
+    this.shareLocaleCode = this.i18n.locale;
+  }
+
+  get availableRoles(): SFProjectRole[] {
+    return SF_PROJECT_ROLES.filter(info => info.canBeShared && this.userShareableRoles.includes(info.role)).map(
+      r => r.role
+    ) as SFProjectRole[];
+  }
+
+  get canUserChangeRole(): boolean {
+    return this.availableRoles.length > 1;
+  }
+
+  get canUserChangeLinkUsage(): boolean {
+    return this.isProjectAdmin;
+  }
+
+  get isLinkReady(): boolean {
+    return this.linkSharingReady && this.sharableLink !== '';
+  }
+
+  get isRecipientOnlyLink(): boolean {
+    return this.shareLinkType === ShareLinkType.Recipient;
+  }
+
+  get projectName(): string {
+    return this.projectDoc?.data?.name ?? '';
+  }
+
+  get sharableLink(): string {
+    return this.projectService.generateSharingUrl(this.linkSharingKey, this.shareLocaleCode.canonicalTag);
+  }
+
+  get shareLinkUsageOptions(): ShareLinkType[] {
+    const options = [];
+    if (this.isProjectAdmin) {
+      options.push(ShareLinkType.Recipient);
+    }
+    if (
+      (this.shareRole === SFProjectRole.CommunityChecker &&
+        this.projectDoc?.data?.checkingConfig.checkingEnabled &&
+        this.projectDoc?.data?.checkingConfig.shareEnabled) ||
+      (this.shareRole !== SFProjectRole.CommunityChecker && this.projectDoc?.data?.translateConfig.shareEnabled)
+    ) {
+      options.push(ShareLinkType.Anyone);
+    }
+    return options;
+  }
+
+  get showLinkSharingUnavailable(): boolean {
+    return !this.pwaService.isOnline;
+  }
+
+  get supportsShareAPI(): boolean {
+    return this.navigator.share != null;
+  }
+
+  copyLink(): void {
+    this.navigator.clipboard.writeText(this.sharableLink).then(async () => {
+      await this.reserveShareLink();
+      await this.noticeService.show(translate('share_control.link_copied'));
```

Comment:
Here's how Google Docs shows the "copied" notice. Note how it's within the dialog, rather than at the bottom of the screen.
![](https://user-images.githubusercontent.com/6140710/211654650-041f1965-79a4-4226-99f6-46e776823d2b.png)